File: lib/commands/http.rb
# frozen_string_literal: true

module Rebbot
  module Commands
    class HTTP < Rebbot::Commands::Base
      on :http, description: 'describe an http status code'

      ANIMALS = %w[
        cat
        dog
      ].freeze

      def with_options(cmd)
        cmd.integer('code', 'status code', required: true)
        cmd.string('animal', 'what animal?', choices: self.ANIMALS.map { |a| [a, a] })
      end

      def on_event(event)
        event.respond(content: url(event.options['code'], event.options['animal']))
      end

      def url(code, animal)
        return "https://http.dog/#{code}.jpg" if animal == 'dog'

        "https://http.cat/#{code}"
      end
    end
  end
end

File: src/commands/http.ts
import { InteractionResponseType } from "discord-interactions";
import { CommandOptionType, cmd } from "../types.js";

export default cmd(
  {
    name: "http",
    description: "Describe an HTTP status code",
    options: [
      {
        name: "code",
        description: "HTTP status code",
        type: CommandOptionType.INTEGER,
        required: true,
      },
      {
        name: "animal",
        description: "Cat or dog?",
        type: CommandOptionType.STRING,
        required: false,
        choices: [
          { name: "cat", value: "cat" },
          { name: "dog", value: "dog" },
        ],
      },
    ],
  },
  async ({ code, animal }) => {
    const url =
      animal === "dog"
        ? `https://http.dog/${code}.jpg`
        : `https://http.cat/${code}`;

    return {
      type: InteractionResponseType.CHANNEL_MESSAGE_WITH_SOURCE,
      data: { content: url },
    };
  },
);
